File: backend/app/schemas/transfer.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TransferOrderOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    inquiry_id: uuid.UUID
    inquiry_no: str
    transfer_status: str
    generated_by: str
    generated_at: datetime
    factory_contract_file: str | None
    finance_transfer_file: str | None
    remark: str | None
    created_at: datetime
    updated_at: datetime


class TransferResponse(BaseModel):
    transfer_id: uuid.UUID
    inquiry_no: str
    factory_contract_url: str
    finance_transfer_url: str
    missing_fields: list[str]
    message: str

File: frontend/src/types/transfer.ts
export interface TransferOrder {
  id: string
  inquiry_id: string
  inquiry_no: string
  transfer_status: "generated" | "regenerated" | "failed"
  generated_by: string
  generated_at: string
  factory_contract_file: string | null
  finance_transfer_file: string | null
  remark: string | null
  created_at: string
  updated_at: string
}

export interface TransferResponse {
  transfer_id: string
  inquiry_no: string
  factory_contract_url: string
  finance_transfer_url: string
  missing_fields: string[]
  message: string
}

export const TRANSFER_STATUS_LABEL: Record<string, string> = {
  generated:   "首次生成",
  regenerated: "重新生成",
  failed:      "生成失败",
}

export const TRANSFER_STATUS_COLOR: Record<string, string> = {
  generated:   "green",
  regenerated: "blue",
  failed:      "red",
}
